sim/shaman/unleash_elements.go

```diff
@@ -0,0 +1,119 @@
+		baseDamage := 1118 + spellCoeff*spell.SpellPower()
+		result := spell.CalcDamage(sim, target, baseDamage, spell.OutcomeMagicHitAndCrit)
+		spell.DealDamage(sim, result)
+	}
+
+	shaman.UnleashElementsFlameTongue = shaman.RegisterSpell(config)
+}
+
+func (shaman *Shaman) registerUnleashElementsFrostbrand(unleashElementsTimer *core.Timer) {
+	//TODO: confirm spell coefficient
+	spellCoeff := 0.40
+
+	config := shaman.newUnleashElementsSpellConfig(unleashElementsTimer)
+	config.SpellSchool = core.SpellSchoolFrost
+	config.ProcMask = core.ProcMaskSpellDamage
+	config.BonusHitRating = float64(shaman.Talents.ElementalPrecision) * core.SpellHitRatingPerHitChance
+
+	//TODO: confirm spell coefficient
+	config.ApplyEffects = func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+		baseDamage := 873 + spellCoeff*spell.SpellPower()
+		result := spell.CalcDamage(sim, target, baseDamage, spell.OutcomeMagicHitAndCrit)
+		spell.DealDamage(sim, result)
+	}
+
+	shaman.UnleashElementsFrostbrand = shaman.RegisterSpell(config)
+}
+
+func (shaman *Shaman) registerUnleashElementsWindfury(unleashElementsTimer *core.Timer) {
+	config := shaman.newUnleashElementsSpellConfig(unleashElementsTimer)
+	config.SpellSchool = core.SpellSchoolPhysical
+	config.ProcMask = core.ProcMaskSpellDamage
+	// TODO: Does this use spell hit?
+	config.BonusHitRating = float64(shaman.Talents.ElementalPrecision) * core.SpellHitRatingPerHitChance
+
+	// TODO: 175% weapon damage and apply buff here
+	// config.ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+	// 	baseDamage := 2298 + spellCoeff*spell.SpellPower()
+	// 	result := spell.CalcDamage(sim, target, baseDamage, spell.OutcomeMagicHitAndCrit)
+	// 	spell.DealDamage(sim, result)
+	// },
+
+	shaman.UnleashElementsWindfury = shaman.RegisterSpell(config)
+}
+
+func (shaman *Shaman) registerUnleashElementsEarthliving(unleashElementsTimer *core.Timer) {
+	spellCoeff := 0.20
+	//TODO: need?
+	//bonusCoeff := 0.02 * float64(shaman.Talents.TidalWaves)
+
+	config := shaman.newUnleashElementsSpellConfig(unleashElementsTimer)
+	config.SpellSchool = core.SpellSchoolNature
+	config.ProcMask = core.ProcMaskSpellHealing
+	config.Flags = core.SpellFlagHelpful
+
+	//TODO: apply buff for 30% on next direct heal
+	config.ApplyEffects = func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+		healPower := spell.HealingPower(target)
+		baseHealing := 1996 + spellCoeff*healPower
+		result := spell.CalcAndDealHealing(sim, target, baseHealing, spell.OutcomeHealingCrit)
+
+		if result.Outcome.Matches(core.OutcomeCrit) {
+			if shaman.Talents.AncestralAwakening > 0 {
+				shaman.ancestralHealingAmount = result.Damage * 0.3
+
+				// TODO: this should actually target the lowest health target in the raid.
+				//  does it matter in a sim? We currently only simulate tanks taking damage (multiple tanks could be handled here though.)
+				shaman.AncestralAwakening.Cast(sim, target)
```

Comment:
We do technically have the raid sim but maintaining it is an absolute nightmare. You could iterate through `shaman.Env.Raid` I believe if you wanted to implement this properly but I don't think it's a huge deal for now

Reply:
this was a comment from a previous dev who put this in. If I work on the healing I will keep that in mind.